beacon-chain/rpc/service.go

```diff
@@ -386,3 +391,21 @@ func (s *Service) logNewClientConnection(ctx context.Context) {
 		}
 	}
 }
+
+// prepareRpcAddressAndProtocol returns a RPC address and protocol.
+// It can be HTTP/S layer or IPC socket, tcp or unix socket.
+func (s *Service) prepareRpcAddressAndProtocol() (string, string, error) {
+	address := fmt.Sprintf("%s:%s", s.cfg.Host, s.cfg.Port)
+	u, err := url.Parse(address)
+	if err != nil {
+		return "", "", err
```

Comment:
Much cleaner (I might be wrong) is to do `return nil, nil, err` or do 
```go
prepareRpcAddressAndProtocol() (address string, protocol string, err error) {
.....
if err != nil {
return
}
....
```

Reply:
I wanted to be compatible with the rest of the implementation of this package but I've decided to change to more elegant and proper way as you have suggested :+1: